storage/remote/queue_manager.go

```diff
@@ -302,11 +341,83 @@ func NewQueueManager(reg prometheus.Registerer, metrics *queueManagerMetrics, lo
 	}
 
 	t.watcher = wal.NewWatcher(reg, wal.NewWatcherMetrics(reg), logger, client.Name(), t, walDir)
+	if t.cfg.SendMetadata {
+		t.metadataWatcher = scrape.NewMetadataWatcher(logger, sm, client.Name(), t, t.cfg.MetadataSendInterval)
+	}
 	t.shards = t.newShards()
 
 	return t
 }
 
+// AppendMetadata queues metadata to be sent to the remote storage.
+func (t *QueueManager) AppendMetadata(metadataList []scrape.MetricMetadata) bool {
+	metadataSet := map[scrape.MetricMetadata]struct{}{}
+
+	for _, metric := range metadataList {
+		metadataSet[metric] = struct{}{}
+	}
```

Comment:
It appears this intermediate map is for deduplication by metric name?  Thats seems unwise - what about metrics from different targets with the same name but different help & type?

Reply:
I think it might be my poor choice of variable naming that's a bit deceiving here, but I'm actually deduping on the whole struct. In this case a combination of metric name, type, help, and unit.

I'll change the variable name to `metadata` instead of metric and add a comment.

Reply:
Hopefully, this naming makes a bit more clear.